File: src/models/Greedy.js
export default class Greedy {
  constructor(graph) {
    this.graph = graph;
  }
  kiemTraMau(z, w, mau) {
    for (var i = 0; i < w; i++) {
      // console.log(mau);
      // console.log(z[i]);
      // console.log(mau === z[i]);
      if (mau === z[i]) {
        return true;
      }
    }
    return false;
  }
  chonMau(g, l) {
    var mau = 1;
    do {
      console.log(g);
      if (!this.kiemTraMau(g, l, mau)) {
        return mau;
        break;
      }
      {
        mau++;
      }
    } while (true);
  }
  toMau() {
    var coloredGrap = {};
    var i = 0,
      j = 1,
      l = 0,
      t = 1,
      mau = 1,
      mauDinh = [],
      dinhTruoc = [],
      mauPhu = [];
    mauDinh[0] = mau;
    dinhTruoc[0] = 0;
    do {
      for (var temp = 0; temp <= l; temp++) {
        if (parseInt(this.graph.logic[j][dinhTruoc[temp]]) === 1) {
          mauPhu[t++] = mauDinh[dinhTruoc[temp]];
        }
      }
      mauDinh[j] = this.chonMau(mauPhu, t);
      l = j;
      dinhTruoc[l] = j;
      j++;
      t = 0;
    } while (j < this.graph.soDinh);
    for (i = 0; i < this.graph.soDinh; i++) {
      coloredGrap[i + 1] = mauDinh[i];
    }
    return coloredGrap;
  }
}

File: src/cppSources/Greedy.cpp
// #include <stdio.h>
// #include <stdlib.h>
// #include <conio.h>

// int a[28][28];
// int n;
// void docfile()
// {
// 	char tenfile[28];
// 	int i, j;
// 	FILE *fp;
// 	printf(" nhap duong dan file:");
// 	gets(tenfile);
// 	fp = fopen(tenfile, "rt");
// 	if (fp == NULL)
// 	{
// 		printf("file khong ton tai!\n");
// 		exit(0);
// 	}
// 	else
// 	{
// 		fscanf(fp, "%d", &n);
// 		for (i = 1; i <= n; i++)
// 			for (j = 1; j <= n; j++)
// 				fscanf(fp, "%d", &a[i][j]);
// 	}
// }

// void xuat()
// {
// 	int i, j;
// 	printf("so dinh:  %d\n", n);
// 	for (i = 1; i <= n; i++)
// 	{
// 		for (j = 1; j <= n; j++)
// 			printf("%d", a[i][j]);
// 		printf("\n");
// 	}
// }
// // kiem tra mau trung
// int kiemtramau(int z[], int w, int mau)
// {
// 	int i;
// 	for (i = 1; i <= w; i++)
// 		if (mau == z[i])
// 			return 1;
// 	return 0;
// }
// // chon mau thich hop de to
// int chonmau(int g[], int l)
// {
// 	int mau = 1;
// 	int i;
// 	do
// 	{
// 		if (kiemtramau(g, l, mau) == 0)
// 		{
// 			return mau;
// 			break;
// 		}
// 		else
// 			mau++;
// 	} while (1);
// }
// // to mau cho cac dinh do thi
// void tomau()
// {
// 	int i, j, l;
// 	int mau, t;
// 	int maudinh[28], dinhtruoc[28], mauphu[28];
// 	i = l = mau = t = 1;
// 	j = 2;
// 	maudinh[1] = mau;
// 	dinhtruoc[1] = 1; // dinh da to dua vao mang dinhtruoc
// 	// mauphu[0]=mau;
// 	do
// 	{
// 		for (int kb = 1; kb <= l; kb++)
// 			if (a[j][dinhtruoc[kb]] == 1)
// 				mauphu[++t] = maudinh[dinhtruoc[kb]];
// 		// dau mau cua dinh da to ma ke voi dinh dang xet vao mang mauphu
// 		maudinh[j] = chonmau(mauphu, t);
// 		l = j;
// 		dinhtruoc[l] = j; // dinh da to dua vao mang dinhtruoc
// 		j++;
// 		t = 0;
// 	} while (j <= n);

// 	for (i = 1; i <= n; i++)
// 		printf("mau dinh%3d: %5d\n", i, maudinh[i]);
// }
// int main()
// {
// 	docfile();
// 	xuat();
// 	tomau();
// }
